lib/compile_to_file/source_file.rb

```diff
@@ -0,0 +1,48 @@
+module CompileToFile
+  class SourceFile
+
+    attr_reader :path
+    attr_reader :swaps
+    attr_reader :skip_requires
+
+    def initialize(path, skip_requires)
+      @path = path
+      @swaps = []
+      @skip_requires = skip_requires
+      yield self if block_given?
+    end
+
+    def swap
+      new_swap = CodeSwap.new
+      swaps << new_swap
+      new_swap
+    end
+
+    def processed_source
+      processed = ""
+
+      File.open(path, "r") do |code_file|
+
+        # Annotate with source file path for debug help
+        processed << "## #{path} ##\n"
+
+        code = code_file.read
+
+        swaps.each { |s| s.apply_to!(code, path) }
+
+        # Strip comments and shebangs
+        code.gsub!(/^\s*#.*\n/, "")
```

Comment:
This could potentially explode with a HEREDOC that uses string interpolation, eg:
```ruby
foo = "bar"
bar = <<EOF
#{foo}
EOF
```

With this gsub `foo` would be an empty string. We could probably change the regex to match any character that isn't a curly brace

```ruby
code.gsub!(/^\s*#[^{]*\n/, "")
```